Content.Server/Light/EntitySystems/MatchstickSystem.cs

```diff
@@ -1,123 +1,47 @@
-        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
-        [Dependency] private readonly SharedAudioSystem _audio = default!;
-        [Dependency] private readonly SharedItemSystem _item = default!;
-        [Dependency] private readonly SharedPointLightSystem _lights = default!;
-        [Dependency] private readonly TransformSystem _transformSystem = default!;
-
-        private readonly HashSet<Entity<MatchstickComponent>> _litMatches = new();
-
-        public override void Initialize()
-        {
-            base.Initialize();
-            SubscribeLocalEvent<MatchstickComponent, InteractUsingEvent>(OnInteractUsing);
-            SubscribeLocalEvent<MatchstickComponent, IsHotEvent>(OnIsHotEvent);
-            SubscribeLocalEvent<MatchstickComponent, ComponentShutdown>(OnShutdown);
-        }
-
-        private void OnShutdown(Entity<MatchstickComponent> ent, ref ComponentShutdown args)
-        {
-            _litMatches.Remove(ent);
-        }
-
-        public override void Update(float frameTime)
-        {
-            base.Update(frameTime);
-
-            foreach (var match in _litMatches)
-            {
-                if (match.Comp.CurrentState != SmokableState.Lit || Paused(match) || match.Comp.Deleted)
-                    continue;
+using Robust.Shared.Timing;
+using Content.Shared.Light.EntitySystems;
 
-                var xform = Transform(match);
+namespace Content.Server.Light.EntitySystems;
 
-                if (xform.GridUid is not {} gridUid)
-                    return;
+public sealed class MatchstickSystem : SharedMatchstickSystem
+{
+    [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+    [Dependency] private readonly TransformSystem _transformSystem = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
 
-                var position = _transformSystem.GetGridOrMapTilePosition(match, xform);
+    public override void Initialize()
+    {
+        base.Initialize();
+    }
 
-                _atmosphereSystem.HotspotExpose(gridUid, position, 400, 50, match, true);
-            }
-        }
+    public override void Update(float frameTime)
+    {
+        var query = EntityQueryEnumerator<MatchstickComponent>();
 
-        private void OnInteractUsing(Entity<MatchstickComponent> ent, ref InteractUsingEvent args)
+        while (query.MoveNext(out var uid, out var match))
         {
-            if (args.Handled || ent.Comp.CurrentState != SmokableState.Unlit)
-                return;
+            if (match.CurrentState != SmokableState.Lit || Paused(uid) || match.Deleted)
+                continue;
 
-            var isHotEvent = new IsHotEvent();
-            RaiseLocalEvent(args.Used, isHotEvent);
+            var xform = Transform(uid);
 
-            if (!isHotEvent.IsHot)
+            if (xform.GridUid is not { } gridUid)
                 return;
```

Comment:
Return means this will terminate the entire method if any single match is not on a grid.